### PR TITLE
refactoring handlers

### PR DESCRIPTION
**Pull request by divyamadala30**
Signed-off-by: Divya Madala <71040191+divyamadala30@users.noreply.github.com>

# Refactored MessageHandlingService

Briefly describe changes proposed in this pull request:
- All research sample(new and updates) are now handled in `ResearchMessageHandlingService`
- All clinical sample(new and updates) are now handled in `ClinicalMessageHandlingService`
- Samples with corrected cmoPatientId are now handled in `PatientCorrectionHandlingService`
---
## Crossing T's and dotting I's

Please follow these checklists to help prevent any unexpected issues from being introduced by the changes in this pull request. If an item does not apply then indicate so by surrounding the line item with `~~` to strikethrough the text. See [basic writing and formatting syntax](https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) for more information.

### I. Web service and data model checklist

Please follow these checks if any changes were made to any classes in the web, service, or persistence layers.

**Code checks:**
- [x] Endpoints were tested to ensure their integrity.
- [ ] ~Screenshots have been provided to demonstrate changes made to the response body JSON schema and/or swagger page.~
- [ ] ~Unit tests were updated in relation to updates to the mocked test data.~

### II. Neo4j models and database schema checklist:
- [ ] ~Neo4j persistence models were changed.~
- [ ] ~The graph database produces the expected changes to models, relationships, and/or property names. [provide screenshot of updated elements in graph db below]~

### III. Message handlers checklist:
- [x] Changes in this PR affect the workflow of incoming messages.
- [x] Messages are following the expected workflow when published to the topic(s) changed or introduced in this pull request.
- [ ] ~Unit tests were added to ensure messages are handled as expected.~

If no unit tests were updated or added, then please explain why: No logic changes, only refactoring

Please describe how the workflow and messaging was tested/simulated:

**Describe your testing environment:**

- NATS [local, **local docker**, dev server, production]
- Neo4j [local, **local docker**, dev server, production]
- MetaDB [local, l**ocal docker**, dev server, production]
- Message publishing simulation [nats cli, docker nats cli, **metadb publisher tool**, other (describe below)]
---
### General checklist:
- [ ] All requested changes and comments have been resolved.
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)

### CODE REVIEW

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/ClinicalMessageHandlingServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    private static volatile boolean shutdownInitiated;
+    private static final ExecutorService exec = Executors.newCachedThreadPool();
+    private static final BlockingQueue<MetadbSample> newClinicalSampleQueue =
+            new LinkedBlockingQueue<MetadbSample>();
+    private static final BlockingQueue<MetadbSample> clinicalSampleUpdateQueue =
+            new LinkedBlockingQueue<MetadbSample>();
+
+    private static CountDownLatch newClinicalSampleHandlerShutdownLatch;
+    private static CountDownLatch clinicalSampleUpdateHandlerShutdownLatch;
+
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/ClinicalMessageHandlingServiceImpl.java:52]: extra line

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/PatientCorrectionHandlingServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    private static final BlockingQueue<Map<String, String>> correctCmoPatientIdQueue =
+            new LinkedBlockingQueue<Map<String, String>>();
+    private static CountDownLatch correctCmoPatientIdShutdownLatch;
+
+    @Override
+    public void initialize(Gateway gateway) throws Exception {
+        if (!initialized) {
+            messagingGateway = gateway;
+            setupCorrectCmoPatientIdHandler(messagingGateway, this);
+            initializeNewMessageHandlers();
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/PatientCorrectionHandlingServiceImpl.java:68]: let's rename this method in all the classes to something like `initializeMessageHandlers()`

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/PatientCorrectionHandlingServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+import org.mskcc.cmo.metadb.service.CrdbMappingService;
+import org.mskcc.cmo.metadb.service.MetadbPatientService;
+import org.mskcc.cmo.metadb.service.MetadbSampleService;
+import org.mskcc.cmo.metadb.service.PatientCorrectionHandlingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PatientCorrectionHandlingServiceImpl implements PatientCorrectionHandlingService {
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/PatientCorrectionHandlingServiceImpl.java:33]: Class rename suggestions:

- CorrectCmoPatientHandlingService
- CorrectCmoPatientHandlingServiceImpl

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java [hunk 4]
```diff
@@ -1,827 +0,0 @@
-package org.mskcc.cmo.metadb.service.impl;
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:1]: Was this class intended to be deleted? It looks like there've been replacements for everything but this class still exists.

**Review by ao508 — CHANGES_REQUESTED** [review]
Initial high-level review - some changes suggested

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java (hunk 4 above)
Comment (divyamadala30) [reply service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:1]: I wanted to rename it to `RequestMessageHandlingServiceImpl`

**Review by ao508 — APPROVED** [review]

**Pull request merged**